Objects/listobject.c

```diff
@@ -3695,6 +3707,14 @@ list_ass_subscript(PyObject* _self, PyObject* item, PyObject* value)
             if (!seq)
                 return -1;
 
+            Py_ssize_t slicelength = adjust_slice_indexes(self, &start, &stop,
+                                                          step);
+
+            if (step == 1) {
+                Py_DECREF(seq);
+                return list_ass_slice(self, start, stop, value);
```

Comment:
I think this does not work if `value` is an iterator. You should use `seq` instead of `value`.

And it seems that there is no direct test for this case. Please add it.

Reply:
Indeed, test was missing! Thanks. Done.